provider/parameter.go

```diff
@@ -170,9 +187,36 @@ func parameterDataSource() *schema.Resource {
 				}
 
 				if parameter.Default != "" {
-					_, defaultIsValid := values[parameter.Default]
-					if !defaultIsValid {
-						return diag.Errorf("default value %q must be defined as one of options", parameter.Default)
+					if parameter.Type == OptionTypeListString && optionType == OptionTypeString {
```

Comment:
This feels ripe to be extracted to a function.

Reply:
I thought so too. I'll do it.